services/venice-server/src/main/java/com/linkedin/venice/listener/ReadQuotaEnforcementHandler.java

```diff
@@ -407,8 +427,20 @@ private void updateQuota(PartitionAssignment partitionAssignment) {
     }
     long quotaInRcu = storeRepository.getStore(Version.parseStoreFromKafkaTopicName(partitionAssignment.getTopic()))
         .getReadQuotaInCU();
-    TokenBucket newStoreBucket = tokenBucketfromRcuPerSecond(quotaInRcu, thisNodeQuotaResponsibility);
-    storeVersionBuckets.put(topic, newStoreBucket); // put is atomic, so this method is thread-safe
+    storeVersionBuckets.compute(topic, (k, v) -> {
+      long newRefillAmount = calculateRefillAmount(quotaInRcu, thisNodeQuotaResponsibility);
+      if (v == null || v.getAmortizedRefillPerSecond() * enforcementIntervalSeconds != newRefillAmount) {
+        // only replace the existing bucket if the difference is greater than 1
+        return tokenBucketfromRcuPerSecond(quotaInRcu, thisNodeQuotaResponsibility);
+      } else {
+        return v;
+      }
+    }); // put is atomic, so this method is thread-safe
```

Comment:
nit: comment is outdated "put"->"compute". Or just remove it as it is kind of obvious